testsuite/tests/singlecluster/gateway/reconciliation/test_gateway_listeners_dns.py

```diff
@@ -0,0 +1,77 @@
+"""Testing specific bug that happens when listener hostname in Gateway gets changed while DNSPolicy is applied."""
+
+import pytest
+from testsuite.gateway.gateway_api.hostname import StaticHostname
+from testsuite.utils import is_nxdomain, sleep_ttl
+
+pytestmark = [pytest.mark.kuadrant_only, pytest.mark.dnspolicy, pytest.mark.tlspolicy]
+
+
+@pytest.fixture(scope="module")
+def wildcard_domain(base_domain, blame):
+    """
+    For this test we want specific domain, not wildcard.
```

Comment:
It shouldn't be called wildcard_domain then. domain/gateway_domain or whatever else you like

Reply:
That is the name of fixture the gateway fixture uses for the hostname. Alternative would be:
- Redefine the whole gateway fixture in the test with the hostname
- Rename the `wildcard_domain` to `gateway_domain` and check this is not conflicting name in different context
- Change the domain with new listener editing functions which I will create as part of work on the issue and move gateway commit to commit fixture to allow for editing before commit

What do you think is best option?

Reply:
The second one. @jsmolar thoughts?

Reply:
The truth is that the name is confusing. However, if renaming leads to some non-trivial refactoring I would vote for a separate PR for that work.

Reply:
I agree, we can solve this as another PR.